### DISCUSSION TITLE
Access to Kubernets API through Teleport with Go app

### DISCUSSION THREAD

**Discussion by morozovcookie** [gravitational/teleport#37843] [Q&A]
Hello everyone!

TokenRequest API was introduced as recommended way to retrieve a token for ServiceAccount since Kubernetes 1.22 or 1.23. But you still able to manually create a secret where will be placed long-lived token. Long-lived token isn't secure, short-lived token generated by TokenRequest API could be accessible only with another token (long-lived, for example).

But as a regular user I am operate with Kubernetes through Teleport installation. So I thought that maybe I could create a special account and authorize inside Kubernetes with Teleport - as I do it as a regular user. But I cannot find this feature in Teleport API.

Is there any way to connect to Kubernetes Cluster from Go application with Teleport API? Basically, I need just need to retrieve a Kubeconfig for my special account. 

Thanks.

**Comment by stevenGravy** [comment]
You can take a look at what the `tsh` client does for `tsh kube login`.

**Reply by morozovcookie** [reply]
I think this is the easiest way to solve my problem :) Thanks!

**Comment by zmb3** [comment]
I'd use [Machine ID](https://goteleport.com/docs/machine-id/access-guides/kubernetes/) to produce a Kubeconfig.

**Reply by morozovcookie** [reply]
Machine ID is the interesting feature, but in my case it will work only with `type: identity`, because I need to select Kubernetes cluster dynamically. But there is a some other cases where I definitely knows which cluster I need to connect. 
I guess I will explore this feature to avoid to build workaround on my own by reverse engineering call of `tctl auth sign`.
